lib/gnat/consumer_supervisor.ex

```diff
@@ -0,0 +1,66 @@
+defmodule Gnat.ConsumerSupervisor do
+  use GenServer
+  require Logger
+
+  def start_link(settings, options) do
+    GenServer.start_link(__MODULE__, settings, options)
+  end
+
+  def init(settings) do
+    Process.flag(:trap_exit, true)
+    {:ok, task_supervisor_pid} = Task.Supervisor.start_link()
+    connection_name = Map.get(settings, :connection_name)
+    subscription_topics = Map.get(settings, :subscription_topics)
+    consuming_function = Map.get(settings, :consuming_function)
+    send self(), :connect
+    state = %{
+      connection_name: connection_name,
+      connection_pid: nil,
+      consuming_function: consuming_function,
+      status: :disconnected,
+      subscription_topics: subscription_topics,
```

Comment:
Sub-topics or transient topics can still be handled by the gnat connection directly (ie setting up temporary inbox topics for replies etc), but this PR is specifically aimed at subscriptions that are being maintained for the lifetime of an application.

In `protobuf-nats` I thought we generated a fixed set of topics at startup and those remained constant? Do we have transient topics for anything other than RPC requests?